src/data/fin/tuple/nat_antidiagonal.lean

```diff
@@ -131,6 +132,31 @@ begin
   refl,
 end
 
+lemma antidiagonal_tuple_pairwise_pi_lex (k n : ℕ) :
+  (antidiagonal_tuple k n).pairwise (pi.lex (<) (λ _, (<))) :=
+begin
+  induction k with k ih generalizing n,
+  { cases n,
+    { simp only [antidiagonal_tuple_zero_zero, list.pairwise_singleton] },
+    { simp only [antidiagonal_tuple_zero_succ, list.pairwise.nil] }, },
+  simp_rw [antidiagonal_tuple, list.pairwise_bind, list.pairwise_map, list.mem_map,
+    forall_exists_index, and_imp, forall_apply_eq_imp_iff₂],
+  simp only [mem_antidiagonal, prod.forall, and_imp, forall_apply_eq_imp_iff₂],
+  simp only [fin.pi_lex_lt_cons_cons, eq_self_iff_true, true_and, lt_self_iff_false, false_or],
+  refine ⟨λ _ _ _, ih _, _⟩,
+  induction n,
+  { rw [antidiagonal_zero],
+    exact list.pairwise_singleton _ _ },
```

Comment:
Defeq (ab)use.
```suggestion
  { exact list.pairwise_singleton _ _ },
```

Reply:
I'd prefer to leave it explicit to match the `antidiagonal_succ` used in the other subgoal. I've added a golf elsewhere instead.